EIPs/eip-0001.md

```diff
@@ -0,0 +1,113 @@
+type: Standards Track
+category: ERC
+status: Draft
+created: 2019-11-28
+---
+
+## Simple Summary
+
+Modification of the `ecrecover` method to restore an Echo account from
+an ECDSA signature.
+
+## Motivation
+
+Minimize the changes necessary when transferring contracts from Ethereum
+to Echo.
+
+## Description of the problem
+
+The solidity `ecrecover` method cannot be used in Echo EVM in the
+version that is assumed by default.
+
+There are three main reasons for this:
+
+##### 1. EdDSA Instead Of ECDSA
+
+Echo uses a different type of signature (EdDSA instead of ECDSA) which
+does not support recovering a public key from a signature. Accordingly,
+the mechanism cannot work by default, since `EcRecovery` returns the
+address who signed the message, and it is impossible to get it from the
+Echo signature
+
+##### 2. Special Address Format
+
+Addresses in Echo EVM is converted account ID. An account with id
+`1.2.100` will have the address
+`0x0000000000000000000000000000000000000064` which cannot be obtained
+from the signature. For this reason, even if the ECDSA signature was
+still supported, we would get problems comparing the sender address and
+the address obtained from the signature. Accordingly, comparison and
+validation (for which the `ecrecover` method is most often used) would
+not work.
+
+##### 3. Authority Mechanism
+
+Since Echo supports an advanced mechanism for setting access rights to
+an account, the private keys of the account can change
+
+## Description of the proposal
+
+The idea is to map the real address of the account and the ECDSA address
+corresponding to the account private key. When calling the contract
+method, which accepts the signature for subsequent verification, use the
+ECDSA signature. Inside the contract, restore the address from the
+signature and use the mapping to get the final address of the account.
+
+## Specification
+
+### General flow
+
+1. User generates a new pair of private and public key ECDSA and the
+   address corresponding to them
+2. User sends the received address to the blockchain, thereby linking
+   the address and an account on the Echo network
+3. User sig data with ECDSA private key and sent to the contract
+4. When using the `ecrecover` method by the contract, Echo determines
+   the address obtained from the signature, finds the address and
+   account match, and returns the address of the Echo account instead of
+   the original address.
+
+### Implementation Details
+
+#### Private Key Generation
+
+User can use absolutely any ECDSA private key. However, to avoid the
+need to store additional private keys, it is proposed to use the
+existing EdDSA account private keys. Keep in mind that multiple accounts
+can use the same set of permissions.
+
+To unify generation and achieve uniqueness for each account, it is
+proposed to use the following option for generating a private key:
+
+```bash
+ECDSAPrivKey = EdDSASig(sha3("1.2.000"), EdDSAPrivKey)
```

Comment:
EdDSA signature is 64 bytes, but EcDSA privkey is 32 bytes logs. Which part of the signature should be used as a key?

Reply:
We can use sha3 of signature as a final private key